### PR TITLE
TropiCataclysm: Plant Zombie additions

### PR DESCRIPTION
**Pull request by Xaritscin**
#### Summary
Mods "Adds some more plant zombies to play against"

#### Purpose of change

Part of the vision of the mod was to add more triffid related content, for this first part i wanted to start by expanding the
Thorny Shambler family which has been mostly untouched in the original mod iterations. 

#### Describe the solution

Added new upgrades ot the Thorny Shambler and one for Zombie Child, all of them based on other varied zombie upgrades with different functions in combat but sharing the same emphasis on being partially plantlike.

#### Describe alternatives you've considered

Game may have already enough issues with Triffids to even consider adding plant aligned zeds.

#### Testing

Ran some test to see if there was any crashes. fast forwarded time to 1 year to see if the new upgrades appeared. found most if not all the new variants on cities among the local hordes

### CODE REVIEW

**Comment by NetSysFire** [comment]
If you want any ideas, maybe add some vegetation-specific zombies. Vine zombies, fern zombies, mossy zombies and zombies with berries growing on them or even hulks with small fruit trees.

File: data/mods/TropiCataclysm/monsters/zed_plant.json [hunk 1]
```diff
@@ -6,6 +6,186 @@
...
+    "name": { "str": "spriteling" },
+    "copy-from": "mon_czombie_thorny",
+    "description": "An undead child seemingly mutated by alien flora.  It has several colorful flowers on its head.",
+    "attack_effs": [ { "id": "paralyzepoison", "duration": 33 } ]
+  },
+  {
+    "id": "mon_boomer_thorny",
+    "type": "MONSTER",
+    "name": { "str": "thorny boomer" },
+    "description": "A fat and slow zombie covered in plant growths of unknown origin.  Some vines seem to extend into its mouth and guts",
```
Comment (BrettDong) [inline data/mods/TropiCataclysm/monsters/zed_plant.json:70]:
```suggestion
    "description": "A fat and slow zombie covered in plant growths of unknown origin.  Some vines seem to extend into its mouth and guts.",
```

**Comment by Xaritscin** [comment]
> If you want any ideas, maybe add some vegetation-specific zombies. Vine zombies, fern zombies, mossy zombies and zombies with berries growing on them or even hulks with small fruit trees.

Right now the plan is to add variants of well known zombies but which have almost the same properties as the Thorny Shambler, that is being covered in alien flora and dropping vegetable materials rather than being fleshy. they also share the same poisonous effects along with some triffid related habilities like spewing pollen or sap.

adding fruit or stuff similar to that im is a bit of an stretch, just having them covered in flowery plants or vines and brambles should be enough.

**Comment by Xaritscin** [comment]
ran some test to see if there was any crashes and it seems i squished all the important bits. submitting for review

**Pull request merged**